oneflow/core/functional/impl/math_functor.cpp

```diff
@@ -1170,7 +1170,17 @@ class SelectTopNFunctor {
   Maybe<TensorTuple> operator()(const TensorTuple& inputs, int32_t n) const {
     MutableAttrMap attr;
     JUST(attr.SetAttr<int32_t>("top_n", n));
+    std::vector<bool> require_grad(inputs.size());
+    std::vector<bool> is_leaf(inputs.size());
+    for (int i = 0; i < n; ++i) {
+      is_leaf.at(i) = (inputs.at(i)->is_leaf());
+      require_grad.at(i) = (inputs.at(i)->requires_grad());
+    }
     const auto& output = JUST(OpInterpUtil::Dispatch<one::TensorTuple>(*op_, inputs, attr));
+    for (int i = 0; i < n; ++i) {
+      inputs.at(i)->set_is_leaf(is_leaf.at(i));
+      inputs.at(i)->set_requires_grad(require_grad.at(i));
```

Comment:
另外注意这个函数的返回值是 `Maybe<void>` 要用 JUST() 包起来。

```suggestion
      JUST(inputs.at(i)->set_requires_grad(require_grad.at(i)));
```